app/models/miq_expression.rb

```diff
@@ -574,180 +574,11 @@ def self._to_ruby(exp, context_type, tz)
   def to_sql(tz = nil)
     tz ||= "UTC"
     @pexp, attrs = preprocess_for_sql(@exp.deep_clone)
-    sql = _to_sql(@pexp, tz) if @pexp.present?
+    sql = to_arel(@pexp, tz).to_sql if @pexp.present?
```

Comment:
This looks good. But, just want to make sure that every case in the case statement in `to_arel` the returned value responds to `to_sql`. Is that correct?